File: leetcode1888/leetcode1888.py
class Solution:
    def minFlips(self, s: str) -> int:
        length = len(s)
        target = "01"
        cnt = 0
        for i in range(0, length):
            cnt += (s[i] != target[i % 2])
        ans = min(cnt, length - cnt)
        for i in range(0, length):
            cnt -= (s[i] != target[i % 2])
            cnt += (s[i] != target[(i + length) % 2])
            ans = min(ans, cnt, length - cnt)
        return ans

File: leetcode1888/leetcode1888.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
    public:
        int minFlips(string s) {
            int len = s.size();
            string target="01";
            int cnt = 0;
            for (int i = 0; i < len; i++){
                cnt += (s[i] != target[i % 2]);
            }
            int ans = min(cnt, len - cnt);
            for (int i = 0; i < len; i ++){
                cnt -= (s[i] != target[i % 2]);
                cnt += (s[i] != target[(i + len) % 2]);
                ans = min({ans, len - cnt, cnt});
            }
            return ans;
        }
    };